client/client.go

```diff
@@ -212,22 +217,10 @@ func (k *serviceModeKeeper) close() {
 }
 
 type client struct {
-	keyspaceID      uint32
-	svrUrls         []string
-	pdSvcDiscovery  *pdServiceDiscovery
-	tokenDispatcher *tokenDispatcher
-
-	// For service mode switching.
-	serviceModeKeeper
-
-	// For internal usage.
-	updateTokenConnectionCh chan struct{}
+	callerID        caller.ID
+	callerComponent caller.Component
 
-	ctx    context.Context
-	cancel context.CancelFunc
-	wg     sync.WaitGroup
-	tlsCfg *tls.Config
-	option *option
+	inner *innerClient
```

Comment:
A nested struct without introducing the field `inner` may reduce the code changes. But I'm not sure whether the pointer here still works, maybe we can have a try.

Reply:
> But I'm not sure whether the pointer here still works, maybe we can have a try.

I tried it before and indeed encountered some issues, so I wrapped the inner client.

Reply:
And I think wrapping `innerclient` helps to reduce the amount of code in client.go, as the code in client.go is a bit excessive.